### ISSUE TITLE
Nuxt Refactor

### ISSUE THREAD

**Issue by guimcaballero** [EmbarkStudios/opensource-website#37]
I'm opening this issue to discuss the possibility to refactor the website to use [Nuxt.js](https://nuxtjs.org/), as talked with @arirawr.

Nuxt is a Vue framework that provides statically generated SPAs with pagination, a nice way to manage Vue Single File Components, and a wide range of [modules](https://github.com/nuxt-community/awesome-nuxt#modules).

[My personal website](https://caballerocoll.com/) is built using Nuxt, and I'm pretty happy with the results. [Here](https://github.com/guimcaballero/blog-template) is a simple blog template as code example.

Pros: 
- Website can be Statically generated, which should be better for performance and SEO. Usually has high [Lighthouse](https://developers.google.com/web/tools/lighthouse/) scores.
- It's a Single Page Application, so it transitions nicely between pages.
- Hot reloading for easy and fast development.
- The [Content](https://content.nuxtjs.org/) module, which is a Git-based Headless CMS.

Cons:
- Adds time to the build step, as the whole website needs to be generated. My personal website takes around 2 minutes to build and deploy on Netlify, vs the 2 seconds it currently takes this website to deploy.
- It definitely adds more JS to be downloaded by the browser, so the final website size would be bigger.
- It stops being a simple HTML + Vue website, so it's added complexity that someone new would have to deal with.

Overall, I don't think it would be needed if the website is to stay this way, but if adding more pages and growing the site is in the roadmap, I think this would be a great improvement.

**Comment by guimcaballero** [comment]
I've had some time this week to make a small prototype of this website using Nuxt. [Here](https://github.com/guimcaballero/embark-nuxt)'s the repo, and [here](https://embark-nuxt.netlify.app/) is the website live in Netlify.

I've just copied all of the html into pages/index.vue, with the script tag at the bottom. All of the components that were in script.js have been moved to components/ in individual Single File Components. All of the features should still work the same.
